src/streams.h

```diff
@@ -509,12 +509,102 @@ class CDataStream
     }
 };
 
+template <typename IStream>
```

Comment:
Perhaps could improve performance, but it'd be trickier to handle streams that are not aligned on 8-byte boundaries. Since the buffer size is an implementation detail, would you be OK leaving that optimization for a later PR?

Reply:
Yeah, let's look at this later. It would complicate the reader indeed. For the writer I think it's pretty straightfoward.

Reply:
Agree with @jimpo, used IStream and OStream should be buffered implementations, not this specific algo.